### PR TITLE
doc: nrf: Added page about Matter ecosystems certification

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
Added dedicated page describing how to certify the Matter product for the specific ecosystem cases. Additionally extended the FFS configuration with the information about the rotating device uid selection.

### CODE REVIEW

File: doc/nrf/protocols/matter/end_product/ecosystems_certification.rst
Comment (LuDuda) [inline doc/nrf/protocols/matter/end_product/ecosystems_certification.rst:13]: I think the word "require" here and in line 21 is a bit too strict (e.g. Google cert is not mandatory afaik). I think we can say that ecosystems offer additional certification programs to ensure that the product works well with the given ecosystem or correctly implement additional proprietary to matter features.

File: doc/nrf/protocols/matter/end_product/ecosystems_certification.rst
Comment (LuDuda) [inline doc/nrf/protocols/matter/end_product/ecosystems_certification.rst:50]: Could we please link to the Amazon certification page? Since it would be nice to understand what "simplified" means - i.e. less testing, no need to send devices to Amazon for lab testing.

File: doc/nrf/protocols/matter/end_product/ecosystems_certification.rst
Comment (LuDuda) [inline doc/nrf/protocols/matter/end_product/ecosystems_certification.rst:54]: What about... Amazon Reference APID -> Reference Design APID ?

**Review by LuDuda — COMMENTED** [review]
Looks great! Small comments below.

Also please do not merge it yet, until we do have a green light from Wi-Fi FFS cert.

File: doc/nrf/protocols/matter/end_product/ecosystems_certification.rst
Comment (kkasperczyk-no) [reply doc/nrf/protocols/matter/end_product/ecosystems_certification.rst:13]: Fair point, applied.

File: doc/nrf/protocols/matter/end_product/ecosystems_certification.rst
Comment (kkasperczyk-no) [reply doc/nrf/protocols/matter/end_product/ecosystems_certification.rst:50]: We link to the Amazon certification page in this sentence `FFS requires passing the additional certification test cases for Matter called `Matter Simple Setup`_.` Do you think it is enough or should we point to some specific part of the page?

File: doc/nrf/protocols/matter/end_product/ecosystems_certification.rst
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/end_product/ecosystems_certification.rst:39]:
```suggestion
The commissioning process is automatically triggered by the Amazon commissioner and it does not require any user interaction. 
The process starts when the commissioner detects the accessory's Bluetooth LE advertising messages in the proprietary format.
```

File: doc/nrf/protocols/matter/end_product/ecosystems_certification.rst
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/end_product/ecosystems_certification.rst:44]:
```suggestion
There are two types of certification:
```

File: doc/nrf/protocols/matter/end_product/ecosystems_certification.rst
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/end_product/ecosystems_certification.rst:47]:
```suggestion
* Default - For standard products or for the creation of a reference design.
* Simplified - For products using a certified reference design.
```

File: doc/nrf/protocols/matter/end_product/ecosystems_certification.rst
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/end_product/ecosystems_certification.rst:50]:
```suggestion
This means that if you create a Matter product based on the |NCS|, you can approach the simplified certification path.
```

File: doc/nrf/protocols/matter/end_product/ecosystems_certification.rst
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/end_product/ecosystems_certification.rst:51]:
```suggestion
You can do this by filling the Reference APID field when you register your product on Amazon's developer page.
Use one of the following values:
```

File: doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst:103]:
```suggestion
Every Matter device must use an unique device identifier for rotating device identifier calculation purpose.
```

File: doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst:104]:
```suggestion
In case of using factory data, the unique identifier is read from the factory data partition.
In other cases, the value set through the :kconfig:option:`CONFIG_CHIP_DEVICE_ROTATING_DEVICE_UID` Kconfig option is used.
```

File: doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst:105]:
```suggestion
You can choose your own unique identifier value to be stored in the factory data or select the auto-generation mode by setting the :kconfig:option:`CONFIG_CHIP_DEVICE_ROTATING_DEVICE_UID` Kconfig option to ``y``.
The auto-generation mode will set the identifier to a random value.
```

File: doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst:106]:
```suggestion
To read more about the factory data generation, see the :ref:`Matter Device Factory Provisioning<ug_matter_device_factory_provisioning>` page.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

File: doc/nrf/protocols/matter/end_product/ecosystems_certification.rst
Comment (kkasperczyk-no) [reply doc/nrf/protocols/matter/end_product/ecosystems_certification.rst:31]: No, I didn't think about that. Fixed.

File: doc/nrf/protocols/matter/end_product/ecosystems_certification.rst
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/end_product/ecosystems_certification.rst:63]: Should these be board names (in which case the `_cpuapp` part should not be included) or build targets (in which case they should be formatted `nrf52840dk_nrf52840`). For refrence from the the Light bulb sample:
image

File: doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst:107]:
```suggestion
You can choose your own unique identifier value to be stored in the factory data or select the auto-generation mode by setting the :kconfig:option:`CONFIG_CHIP_DEVICE_ROTATING_DEVICE_UID` Kconfig option to ``y``.
In case of using factory data, the unique identifier is read from the factory data partition.
Otherwise, the auto-generation mode will set the identifier to a random value.
```
Mulled that around in my head for a bit more. Check that no relevant information was lost in the reordering.

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

File: doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst
Comment (kkasperczyk-no) [reply doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst:107]: To be honest I think the previous version was better. The information we want to include is that you can store the unique id in the factory data or use hardcoded version from the firmware. Next, if you selected factory data, there are again two ways to go. You can define your own value and store it in the factory data, or you don't care about it and choose the auto-generation that will use some random value and save it in the factory data.

The suggested version starts from: 

> You can choose your own unique identifier value to be stored in the factory data or select the auto-generation mode by setting the ..

and in my opinion it informs that I can store id in the factory data or select auto-generation, so I would assume the latter will not use factory data, which is not true. 

What do you think?

Comment (wiba-nordic) [reply doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst:107]: Ah, so the auto-generation also stores the generated identifier in the factory data, and that is where it is read from regardless?

Comment (kkasperczyk-no) [reply doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst]: Yes, btw. I think I made some mistake and used the same kconfig name two times. Let me explain it one more time. the user can choose between:
a. Keeping unique id, as a constant value in the firmware (passed through `CONFIG_CHIP_DEVICE_ROTATING_DEVICE_UID` Kconfig)
b. Keeping unique id in factory data

In case of selecting b, we have another two options:
a. Select auto-generation that will use random unique id value and put it into the storage (`CONFIG_CHIP_DEVICE_GENERATE_ROTATING_DEVICE_UID` equal to `y`, then `CONFIG_CHIP_DEVICE_ROTATING_DEVICE_UID` is ignored)
b. Force using own unique id value that will be put into the storage (`CONFIG_CHIP_DEVICE_GENERATE_ROTATING_DEVICE_UID` equal to `n`, then `CONFIG_CHIP_DEVICE_ROTATING_DEVICE_UID` is used)

File: doc/nrf/protocols/matter/end_product/ecosystems_certification.rst
Comment (kkasperczyk-no) [reply doc/nrf/protocols/matter/end_product/ecosystems_certification.rst]: Fair point, actually I think it should be hardware platforms. I updated the table with this change.

**Comment by kkasperczyk-no** [comment]
@LuDuda as discussed F2F I removed Wi-Fi APID, so I'm revoing DNM label.

**Review by markaj-nordic — APPROVED** [review]
LGTM, there is however a licensing warning reported on the first altered file.

**Comment by kkasperczyk-no** [comment]
> LGTM, there is however a licensing warning reported on the first altered file.

I'm not sure whether this is really related to my change.

File: doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst:108]:
```suggestion
By default, the identifier is set to a random value and stored in the factory data partition.
You can choose your own unique identifier value instead by setting the :kconfig:option:`CONFIG_CHIP_DEVICE_GENERATE_ROTATING_DEVICE_UID` Kconfig option to ``n`` and using the :kconfig:option:`CONFIG_CHIP_DEVICE_ROTATING_DEVICE_UID` Kconfig option.
When using your own identifier, the value can be stored in either firmware or factory data.
For more information about the factory data generation, see the :ref:`Matter Device Factory Provisioning<ug_matter_device_factory_provisioning>` page.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

**Review by wiba-nordic — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]
Looks great!

Re-adding DNM, awaiting one confirmation.

**Pull request merged**
